braid/base.c

```diff
@@ -321,6 +412,41 @@ _braid_BaseSum(braid_Core        core,
    _braid_CoreFcn(core, sum)(app, alpha, x->userVector, beta, y->userVector);
    _braid_CoreElt(core, timer_user_sum) += _braid_MPI_Wtime(core) - timer;
 
+   /* Sum over the basis vectors */
+   if ( x->basis && y->basis )
+   {  /* Add together the bases if they both exist */
+      _braid_BaseSumBasis(core, app, alpha, x->basis, beta, y->basis);
+   }
+   else if ( y->basis )
```

Comment:
Just out of curiosity, do these cases of (x->basis = Null), but  (y->basis not Null) occur?

Reply:
This is a little over-engineered at the moment... I wrote it this way before I had finished the communication, and I wasn't sure that I would always be sending the basis in every communication, in which case the vector constructed by BufUnpack would have x->basis == Null. However, I think the way things are now this should never happen.